src/crimson/osd/osd.h

```diff
@@ -126,6 +127,15 @@ class OSD : public ceph::net::Dispatcher,
                                        version_t last,
                                        Ref<MOSDMap> m);
 
+  // order the promises in descending order of the waited osdmap epoch,
+  // so we can access all the waiters expecting a map whose epoch is less
+  // than a given epoch
```

Comment:
does the shared_promise guarantee the wait execute in order?  e.g several places call get_shared_future in a certain order, when shared_promise fulfill the shared_future,  the continuations execute the same order as get_shared_future or not?

Reply:
the order of wakeup is not determined by `shared_promise`, it's decided by `OSD::consume_map()`. see the commit message of c087b1ce369012646451b47b335cd4a032b74a4e .